### ISSUE TITLE
GApps on uncertified devices

### ISSUE THREAD

**Issue by lukechadwick** [microg/android_packages_apps_GmsCore#510]
https://www.xda-developers.com/google-blocks-gapps-uncertified-devices-custom-rom-whitelist/

Is this new google certification block going to affect us microg users?

**Comment by mar-v-in** [comment]
Maybe, but if it does, there will be a way to workaround ;)

**Comment by ArchangeGabriel** [comment]
@Nanolx We are also tracking this in #482.

Apparently Google started to enforce device certification y-day, and pushed an update to DroidGuard at the same time. Now, in order to use PlayServices, you either have to pass SafetyNet (which μG does not currently), or register your device as modified (which none of us want to do I suppose).

I suspect that SafetyNet will thus become mandatory to use μG, and that μG will have to be modified to account for PlayServices requiring this SafetyNet check. But @mar-v-in will now better for sure, however in the mean time we are in the dark.

**Comment by lukechadwick** [comment]
Perhaps we can spoof the build date if this is still relevant/true:

"This change apparently went into effect March 16th and affects any software builds made after this date (Google Play Services checks ro.build.fingerprint for the build date apparently)."

**Comment by Benutzer1234** [comment]
My firmware and all system relevant files are from 2016 and I didn't update them. MicroG on this phone stopped working yesterday. An old date won't help us, even if it is real.

**Comment by ArchangeGabriel** [comment]
@lukechadwick I think this was true until y-day, but they changed something in their policy at that point.

**Comment by drrossum** [comment]
I have used my phone without google play store for a year now and am quite happy with it. I download an individual apk here and there in case I really want to update an app. Not having the temptation to constantly update all the apps gives a peace of mind!

Not saying that it would not be nice to try to fix this, of course!

**Comment by Nanolx** [comment]
Yes, it does no longer check the build date, my Nexus 6 has a ROM older than 16.03.2018, aswell.

Also registering the GSF-ID does **not** solve the issue (since microG randomly spoofs one which gets lost upon factory reset, there's no harm testing it), as it would when using GApps.

While you can of course still use Yalp Store, despite **CTS** status, many other things won't work, though, because microG no longer even attests **BasicIntegrity** successfully since yesterday, many apps no longer work with microG.

This is very much a show-stopper, since SafetyNet gets used by more and more apps (banking, games, multimedia), way more than just Google-Stuff (despite often being uselessley implemented in first place: why does Nintendo implement SafetyNet check in Super Mario Run? It's a **paid** game!?).

**Comment by Benutzer1234** [comment]
@ArchangeGabriel
I doubt that G. started to enforce device certification yesterday.
For me it looks more like they just updated DroidGuard.
See the first edit at https://github.com/microg/android_packages_apps_GmsCore/issues/510#issuecomment-379859128

**Comment by ale5000-git** [comment]
microG now pass SafetyNet and it also has "Device profiles" to allow you to spoof a certified device.
